src/Examples/Tools/DegeneracyHunter/degeneracy_hunter.ipynb

```diff
@@ -0,0 +1,1538 @@
+      "Complementarity.........:   2.6658012082875325e-09    2.6658012082875325e-09\n",
+      "Overall NLP error.......:   2.6658012082875325e-09    2.6658012082875325e-09\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 23\n",
+      "Number of objective gradient evaluations             = 18\n",
+      "Number of equality constraint evaluations            = 23\n",
+      "Number of inequality constraint evaluations          = 23\n",
+      "Number of equality constraint Jacobian evaluations   = 18\n",
+      "Number of inequality constraint Jacobian evaluations = 18\n",
+      "Number of Lagrangian Hessian evaluations             = 17\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.004\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n",
+      "x[ 1 ]= 5.061595738300216\n",
+      "x[ 2 ]= 9.184026563775058e-26\n",
+      "x[ 3 ]= -3.4867300513803765\n"
+     ]
+    }
+   ],
+   "source": [
+    "opt.options['max_iter'] = 50\n",
+    "opt.solve(m2b, tee=True)\n",
+    "\n",
+    "for i in m2b.I:\n",
+    "    print(\"x[\",i,\"]=\",m.x[i]())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We get the same answer as before, but careful inspection of the Ipopt output reveals a subtle improvement. Notice `ls` is only 1 or 2 for all of the iterations, in contrast to more than 20 for the original model. This means Ipopt is taking (nearly) full steps for all iterations.\n",
+    "\n",
+    "Let's also compare the number of function evaluations.\n",
+    "\n",
+    "Original model (using Ipopt 3.13.2 with `ma27`):\n",
+    "```\n",
+    "Number of objective function evaluations             = 111\n",
+    "Number of objective gradient evaluations             = 17\n",
+    "Number of equality constraint evaluations            = 111\n",
+    "Number of inequality constraint evaluations          = 111\n",
+    "Number of equality constraint Jacobian evaluations   = 17\n",
+    "Number of inequality constraint Jacobian evaluations = 17\n",
+    "Number of Lagrangian Hessian evaluations             = 16\n",
+    "```\n",
+    "\n",
+    "Reformulated model (using Ipopt 3.13.2 with `ma27`):\n",
+    "```\n",
+    "Number of objective function evaluations             = 23\n",
+    "Number of objective gradient evaluations             = 18\n",
+    "Number of equality constraint evaluations            = 23\n",
+    "Number of inequality constraint evaluations          = 23\n",
+    "Number of equality constraint Jacobian evaluations   = 18\n",
+    "Number of inequality constraint Jacobian evaluations = 18\n",
+    "Number of Lagrangian Hessian evaluations             = 17\n",
+    "```\n",
+    "\n",
+    "Removing a **single redundant constraint** reduced the number of objective and constraint evaluations by a **factor of 5**!\n"
```

Comment:
Whilst the implications of this should be fairly obvious, it might not hurt to just emphasize that this directly relates to solver time and that it will only get worse with more complex problems (not to mention that the problem might fail to converge at all).

Reply:
Good suggestion.